src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/DoNotCheckFlagFromDifferentEnumTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Usage.DoNotCheckFlagFromDifferentEnum,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Usage.DoNotCheckFlagFromDifferentEnum,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Usage.UnitTests
+{
+    public class DoNotCheckFlagFromDifferentEnumTests
+    {
+        [Fact]
+        public async Task CA2248_EnumTypesAreDifferent_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+public class C
+{
+    [Flags]
+    public enum MyEnum { A, B, }
+
+    [Flags]
+    public enum OtherEnum { A, }
+
+    public void Method(MyEnum m)
+    {
+        [|m.HasFlag(OtherEnum.A)|];
+    }
+}");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Public Class C
+    <Flags>
+    Public Enum MyEnum
+        A
+        B
+    End Enum
+
+    <Flags>
+    Public Enum OtherEnum
+        A
+    End Enum
+
+    Public Sub Method(ByVal m As MyEnum)
+        [|m.HasFlag(OtherEnum.A)|]
+    End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public async Task CA2248_EnumTypesAreSame_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+public class C
+{
+    [Flags]
+    public enum MyEnum { A, B, }
+
+    public void Method(MyEnum m)
+    {
+        m.HasFlag(MyEnum.A);
+    }
+}");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Public Class C
+    <Flags>
+    Public Enum MyEnum
+        A
+        B
+    End Enum
+
+    Public Sub Method(ByVal m As MyEnum)
+        m.HasFlag(MyEnum.A)
+    End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public async Task CA2248_EnumTypesAreSameButNotFlag_NoDiagnostic()
```

Comment:
Can you also add a test where enum types are different by neither enum types have flags attribute to complete the test matrix?

Reply:
It seems runtime throws an exception when comparing different non-flags enum types as well. So this rule should not have any dependence on whether the enum is marked as Flags or not (which seems to be the case in rule implementation, but not Title)

```csharp
using System;

public class C
{
    public enum MyEnum { A, B, }
    public enum OtherEnum { A, }

    static void Main()
    {
        Console.WriteLine(MyEnum.A.HasFlag(OtherEnum.A));
    }
}
```

Throws the same unhandled exception at runtime.

Reply:
Done! Thanks for catching-up this case.